### PR TITLE
fix: set link default value to random

### PR DESCRIPTION
**Pull request by pierrbt**
The link input has now a default value, which is a random string.
I've kept the Generate link to make another one, if the user wants, or if he make a bad manipulation and have to generate another link.

### CODE REVIEW

**Comment by pierrbt** [comment]
This Pull request resolve #180

File: package.json [hunk 1]
```diff
@@ -8,5 +8,8 @@
     "release:patch": "cd backend && npm version patch --commit-hooks false && cd ../frontend && npm version patch --commit-hooks false && cd .. && git add . && npm version patch --force -m 'release: %s' && git push && git push --tags",
     "release:minor": "cd backend && npm version minor --commit-hooks false && cd ../frontend && npm version minor --commit-hooks false && cd .. && git add . && npm version minor --force -m 'release: %s' && git push && git push --tags",
     "deploy:dev": "docker buildx build --push --tag stonith404/pingvin-share:development --platform linux/amd64,linux/arm64 ."
+  },
+  "dependencies": {
```
Comment (stonith404) [inline package.json:12]:
```suggestion
  "devDependencies": {
```

I would add add it to the dev dependencies.

**Review by stonith404 — APPROVED** [review]

**Comment by stonith404** [comment]
Nice, thanks again for all your PRs :)

**Comment by pierrbt** [comment]
No problems, I'm working on the reverse shares options.

**Comment by pierrbt** [comment]
@stonith404 

I'll just need help because I had to modify a bit the database, to add a string, which contains a JSON object : 

```ts
export type ReverseShareOptions = {
  easyMode: boolean;

  customLinkEnabled: boolean;
  passwordEnabled: boolean;
  descriptionEnabled: boolean;
  maximalViewsEnabled: boolean;
}
```

And I also changed the backend service and DTOs, but I would like to add this modification, without breaking the current db, how can I do that ? I would like to put default values for all the current reverse shares but I don't know how to do ...

**Comment by pierrbt** [comment]
For the frontend there is no problem, because I can do smth like this : 
```ts
const getMyReverseShares = async (): Promise<MyReverseShare[]> => {
  const shares = (await api.get("reverseShares")).data;
  for (const share of shares)
  {
    share.sharesOptions = JSON.parse(share.sharesOptions);
    if(Object.keys(share.sharesOptions).length === 0)
      share.sharesOptions = structuredClone(defaultReverseShareOptions);
  }

  return shares;
};
```

But the backend db resets if the schema has changed. Personally, I have regenerated the db : 

```
npx prisma db push
npx prisma db seed
```

But for all the other users, how could I make the new release without having to reset their db ?

**Comment by stonith404** [comment]
@pierrbt I probably would add a new table and add a 1:1 connection between the `ReverseShare` and `ReverseShareOptions` table as a a relational database shouldn't have a JSON object inside a cell.

For the migration part, it is quite easy with prisma. You just have to run `npx prisma dev`. Then a sql file will get created in `./prisma/migrations/{your_migration}`. Now you can update the sql file that the existing reverse shares will get the correct data.

Btw. you can write me on Discord (stonith404) then we don't have to discuss things in random PRs or issues.

**Pull request merged**